kallsyms/kallsyms.go

```diff
@@ -601,20 +605,104 @@ func (s *Symbolizer) pollKobjectClient(kobjectClient *kobject.Client) {
 			// Notify worker thread without blocking
 			select {
 			case s.reloadModules <- true:
+			case <-ctx.Done():
+				return
 			default:
 			}
 		}
 	}
 }
 
+// bpfKsymsTrigger attaches a minimal eBPF program to bpf events, that update
+// kallsyms, to notify user space about changes.
+func (s *Symbolizer) bpfKsymsTrigger(ctx context.Context) {
+	events, err := ebpf.NewMap(&ebpf.MapSpec{
+		Type: ebpf.PerfEventArray,
+		Name: "ksym_update_trigger",
+	})
+	if err != nil {
+		log.Errorf("Failed creating perf event array: %s", err)
+		return
+	}
+	defer events.Close()
+
+	rd, err := perf.NewReader(events, os.Getpagesize())
+	if err != nil {
+		log.Errorf("Failed creating event reader: %s", err)
+		return
+	}
+	defer rd.Close()
+
+	var progSpec = &ebpf.ProgramSpec{
+		Name:    "ksym_update_trigger",
+		Type:    ebpf.Kprobe,
+		License: "GPL",
+	}
+
+	// Minimal eBPF program that writes the static value '1' to the perf event
```

Comment:
Isn't it better for clarity if we move this to eBPF C?

Reply:
friendly ping for feedback @fabled and @christos68k

Reply:
This does not reflect what was I described few messages above. Most importantly this still makes kallsyms depend on ebpf stuff directly. Do you have any rationale on why you chose this approach contrary to what was requested?